src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hot_node/ThreadMetricsRca.java

```diff
@@ -0,0 +1,375 @@
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.opensearch.performanceanalyzer.PerformanceAnalyzerApp;
+import org.opensearch.performanceanalyzer.grpc.FlowUnitMessage;
+import org.opensearch.performanceanalyzer.metrics.AllMetrics;
+import org.opensearch.performanceanalyzer.metricsdb.MetricsDB;
+import org.opensearch.performanceanalyzer.rca.framework.api.Metric;
+import org.opensearch.performanceanalyzer.rca.framework.api.Rca;
+import org.opensearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import org.opensearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import org.opensearch.performanceanalyzer.rca.framework.api.metrics.Thread_Blocked_Time;
+import org.opensearch.performanceanalyzer.rca.framework.api.metrics.Thread_Waited_Time;
+import org.opensearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import org.opensearch.performanceanalyzer.rca.framework.core.RcaConf;
+import org.opensearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import org.opensearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+
+/**
+ * Analyzes thread_blocked_time and thread_waited_time metrics for all threads and in specific
+ * transport threads and publishes the following data: 1. Number of threads which are blocked for
+ * more than 5 seconds in past minute. 2. Number of threads which are in waiting/timed-waiting state
+ * for more than 5 seconds in past minute. 3. Maximum value of thread_blocked_time for a single
+ * thread in the past minute. 4. Maximum value of thread_waited_time for a single thread in the past
+ * minute.
+ */
+public class ThreadMetricsRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
+
+    private static final Logger LOG = LogManager.getLogger(ThreadMetricsRca.class);
+    public static final double HIGH_BLOCKED_TIME_THRESHOLD_IN_SECONDS = 5d;
+    private final int rcaPeriod;
+    private final Thread_Blocked_Time threadBlockedTime;
+    private final Thread_Waited_Time threadWaitedTime;
+    private final Clock clock;
+    @VisibleForTesting final List<ThreadAnalysis> threadAnalyses;
+
+    public ThreadMetricsRca(
+            final Thread_Blocked_Time threadBlockedTime,
+            final Thread_Waited_Time threadWaitedTime,
+            final int rcaPeriodInSeconds) {
+        super(rcaPeriodInSeconds);
+        this.rcaPeriod = rcaPeriodInSeconds;
+        this.threadBlockedTime = threadBlockedTime;
+        this.threadWaitedTime = threadWaitedTime;
+        threadAnalyses = new ArrayList<>();
+        initThreadAnalyses();
+        this.clock = Clock.systemUTC();
+    }
+
+    private void initThreadAnalyses() {
+        // all threads
+        threadAnalyses.add(
+                new ThreadAnalysis(
+                        s -> true,
```

Comment:
Wouldn't this also include transport threads as well which are already defined below and calculate twice for it? We can switch this off for now, as needed we can add more threads here.

Reply:
Sure. removed the generic analysis and retained only transport thread analysis for now. Agreed that we can add more threadpools to analysis if deemed necessary.